packages/grafana-schema/src/common/data_frame.cue

```diff
@@ -0,0 +1,216 @@
+package common
+
+//  See also:
+//  https://github.com/grafana/grafana-plugin-sdk-go/blob/main/data/frame_type.go
+DataFrameType:
```

Comment:
Saw it used in the [dashboard datasource](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/dashboard/types.ts#L1), but not sure how important that is.

Reply:
I think I would need more of the larger context to understand. Is the goal here to use cue just for typescript and the frontend, or do we want to be generating the Go types that go with `data.Frame` (field, etc) as well?

If it is the case we are going for the Go types as well, I think this is going to be hard one. Currently the frame code and libraries live in another repo, https://github.com/grafana/grafana-plugin-sdk-go/tree/main/data . This is also a place we want a tight binding, and is also generally performance sensitive.

So perhaps continue for the sake of exploration (to learn just what the set of issues will be for end-to-end type gen and binding here, but maybe won't be mergable for some time)? Also perhaps limit to just FrameMeta? I expect that is easier.

Maybe @ryantxu has thoughts?

Reply:
> Currently the frame code and libraries live in another repo, https://github.com/grafana/grafana-plugin-sdk-go/tree/main/data (and this library is imported by virtually every data source). This is also a place we want a tight binding

Tight binding meaning, careful control over when these types change and keeping things in sync, right? That's been my assumption to date, and why i've assumed there would be a couple additional steps before we could bring these into canonical schema systems.

But yes, at some point in the medium-term future, it looks like we'll need to get these into sync.

Reply:
> Tight binding meaning, careful control over when these types change and keeping things in sync, right?

Yup! So a great candidate for bring things together - but a harder one perhaps.

Reply:
I'll leave the schema for DataFrame here, in case anyone will want to continue working on it in the future.

Reply:
> but a harder one perhaps.

Could be. For now, i don't see a significant difference vs. what we have to implement with grok - synchronization from the SSoT in grafana out to a second repo on a release-tied cadence. We've got a decent plan in place for that (cc @malcolmholmes) for grok, and my handwavy expectation has been to reuse as much of we can of that approach in the plugin sdk.

Reply:
This will be super interesting to see what we land with, but note that is is a BIG project that is already actively used by every backend plugin and most frontend plugins.  The handwavy generation logic will likely need to generate exactly what exists (or very close with clear depreciation path) before it is useable directly.

But yes conceptually it is the same (actually simpler than) dashbaords